No known cycle-counting instruction on ppc64

`mirage-crypto` fails to install on ppc64:

```
#=== ERROR while compiling mirage-crypto.0.8.1 ================================#
# context              2.0.7 | linux/ppc64 | ocaml-base-compiler.4.10.0 | file:///home/opam/opam-repository
# path                 ~/.opam/4.10/.opam-switch/build/mirage-crypto.0.8.1
# command              ~/.opam/4.10/bin/dune build -p mirage-crypto -j 79
# exit-code            1
# env-file             ~/.opam/log/mirage-crypto-8-9e20af.env
# output-file          ~/.opam/log/mirage-crypto-8-9e20af.out
### output ###
#       ocamlc src/entropy_cpu_stubs.o (exit 2)
# (cd _build/default/src && /home/opam/.opam/4.10/bin/ocamlc.opt -g -I /home/opam/.opam/4.10/lib/bigarray-compat -I /home/opam/.opam/4.10/lib/cstruct -I /home/opam/.opam/4.10/lib/eqaf -I /home/opam/.opam/4.10/lib/eqaf/bigstring -I /home/opam/.opam/4.10/lib/eqaf/cstruct -ccopt -O2 -ccopt -fno-strict-aliasing -ccopt -fwrapv -ccopt -fPIC -ccopt --std=c99 -ccopt -Wall -ccopt -Wextra -ccopt -Wpedantic -ccopt -O3 -ccopt -g -o entropy_cpu_stubs.o native/entropy_cpu_stubs.c)
# native/entropy_cpu_stubs.c: In function 'caml_cycle_counter':
# native/entropy_cpu_stubs.c:86:2: error: #error ("No known cycle-counting instruction.")
#  #error ("No known cycle-counting instruction.")
#   ^~~~~
# native/entropy_cpu_stubs.c:88:1: warning: control reaches end of non-void function [-Wreturn-type]
```

Any suggestions on how to get this working?

--- a/CHANGES.md
+++ b/CHANGES.md
@@ -1,3 +1,5 @@
+* Add support for ppc64le. (@avsm)
+
 ## v0.8.1 (2020-07-02)
 
 * Add Chacha20 implementation (based on abeaumont/ocaml-chacha), supporting

--- a/CHANGES.md
+++ b/CHANGES.md
@@ -1,4 +1,6 @@
-* Add support for ppc64le. (@avsm)
+## v0.8.2 (2020-07-25)
+
+* Add support for ppc64le. (#76 @avsm)
 
 ## v0.8.1 (2020-07-02)
 

--- a/src/native/entropy_cpu_stubs.c
+++ b/src/native/entropy_cpu_stubs.c
@@ -82,6 +82,10 @@ CAMLprim value caml_cycle_counter (value __unused(unit)) {
   return Val_long (__rdtsc ());
 #elif defined (__arm__) || defined (__aarch64__)
   return Val_long (read_virtual_count ());
+#elif defined(__powerpc64__)
+    uint64_t rval;
+    __asm__ volatile ("mfspr %0, 268":"=r" (rval));
+    return rval;
 #else
 #error ("No known cycle-counting instruction.")
 #endif

--- a/src/native/ghash_generic.c
+++ b/src/native/ghash_generic.c
@@ -16,7 +16,7 @@
  * !LARGE_TABLES -> 8K per key, ~3x slower. */
 #define __MC_GHASH_LARGE_TABLES
 
-#if defined (__x86_64__) || defined (__aarch64__)
+#if defined (__x86_64__) || defined (__aarch64__) || defined(__powerpc64__)
 
 #define __set_uint128_t(w1, w0) (((__uint128_t) w1 << 64) | w0)
 
@@ -103,4 +103,4 @@ mc_ghash_generic (value m, value hash, value src, value off, value len) {
   return Val_unit;
 }
 
-#endif /* x86_64 || aarch64 */
+#endif /* x86_64 || aarch64 || powerpc64 */

--- a/src/native/poly1305-donna.c
+++ b/src/native/poly1305-donna.c
@@ -7,7 +7,7 @@ typedef struct poly1305_context {
         unsigned char opaque[136];
 } poly1305_context;
 
-#if defined (__x86_64__) || defined (__aarch64__)
+#if defined (__x86_64__) || defined (__aarch64__) || defined(__powerpc64__)
 #include "poly1305-donna-64.h"
 #else
 #include "poly1305-donna-32.h"

